### PR TITLE
Issue #19040: Remove Leanplum (Nightly)

### PR DESCRIPTION
**Pull request by pocmo**
This PR removes Leanplum from the codebase in the `master` branch (Nightly, 90.0).

* This patch does not update the settings string containing "Leanplum" yet. It's still to be discussed what the right string has to be.
* @jonalmeida Can you take a look at the push changes in this PR?
* The patch applies to beta and release too. I'll open separate PRs once this has gone through a first review and automation is happy.
* Please review, but do not land yet since this needs a bit of coordination.

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/push/FirebasePushService.kt [hunk 1]
```diff
@@ -5,64 +5,12 @@
...
 import android.annotation.SuppressLint
-import com.google.firebase.messaging.RemoteMessage
-import com.google.firebase.messaging.FirebaseMessagingService
-import com.leanplum.LeanplumPushFirebaseMessagingService
-import com.leanplum.LeanplumPushService
-import mozilla.components.concept.push.PushService
-import mozilla.components.lib.push.firebase.AbstractFirebasePushService
 import mozilla.components.feature.push.AutoPushFeature
-
-/**
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/fenix/push/FirebasePushService.kt:16]: 🔥

**Review by jonalmeida — APPROVED** [review]
> @jonalmeida Can you take a look at the push changes in this PR?

So good to see that hack we wrote for push going away! 😄

File: app/src/main/java/org/mozilla/fenix/components/BackgroundServices.kt [hunk 2]
```diff
@@ -217,13 +217,11 @@ internal class TelemetryAccountObserver(
             metricController.track(it)
         }
 
-        // Used by Leanplum as a context variable.
         settings.fxaSignedIn = true
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/fenix/components/BackgroundServices.kt:220]: Should we remove these settings as well?

Comment (pocmo) [reply app/src/main/java/org/mozilla/fenix/components/BackgroundServices.kt:220]: At first I removed it and then I was wondering whether we would want to keep it for other things. But maybe you are right, no need to keep for now, I'll get rid of it.

**Comment by pocmo** [comment]
There are no active Leanplum campaigns on Nightly anymore. This can land. We can deal with the settings string in a follow-up once a decision has been made.

**Pull request merged**